multi_import/formats.py

```diff
@@ -130,6 +131,32 @@ def pre_read(self, file_object):
         file_object = strings.normalize_string(file_object)
         return file_object
 
+    def detect(self, file_handler, file_contents):
+        if self.is_valid_csv(file_handler, file_contents):
+            # Handles row validations
+            return super(CsvFormat, self).detect(
+                file_handler,
+                file_contents
+            )
+        return False
+
+    def is_valid_csv(self, file_handler, file_contents):
+        file_object = self.get_file_object(file_handler, file_contents)
+        try:
+            # Would error out if invalid csv file
+            Dataset().load(file_object, "csv")
+            # Note: dataset is valid for test_file.yaml in the tests directory
+            # Would need suggestions on how to better handle this
+            return not YamlFormat().detect(file_handler, file_contents)
+        except (
+            InvalidDimensions,
+            UnsupportedFormat,
+            AttributeError,
+            _csvError
+        ):
+            pass
```

Comment:
Is there anything else we want to do with the exception handling here besides passing silently? Would there be any value in logging details from the exceptions to understand why they're occurring?

Reply:
The errors essentially address the other file format detection (xlsx, yaml etc.) errors when the Dataset load method is called. If we log the details, we would be logging if a non-csv file type was being processed. The original detect method that was being used also [passed](https://github.com/sdelements/django-multi-import/pull/37/files#diff-0f06514317fb4dd1e8489144b7d602007245a6799e4c3ede86f9b79b71153723L65) the errors.  In the other format detection functions, either a pass or False value is returned.